traitsui/qt4/tabular_editor.py

```diff
@@ -266,11 +266,13 @@ def dispose(self):
             self.refresh_editor, "adapter.+update", remove=True
         )
         self.on_trait_change(
-            self.update_editor, "adapter.columns", remove=True
+            self._adapter_columns_updated, "adapter.columns", remove=True
         )
 
         self.adapter.cleanup()
 
+        toolkit.destroy_control(self.control)
```

Comment:
Backtracking on this: I need to think about harder about why this fixes the error. It is strange that we should need to call `destroy_control` here, because I'd expect the UI to do this for all the editors, and in fact, `traitsui.ui.UI.finish` calls this.  I will resurrect the workaround because that is a less risky thing to do, and would make it easier for this PR to be backported to 7.0.1 bug fix release.